spring-kafka/src/test/java/org/springframework/kafka/retrytopic/RetryTopicClassLevelIntegrationTests.java

```diff
@@ -0,0 +1,849 @@
+					.maxAttempts(5)
+					.concurrency(1)
+					.useSingleTopicForSameIntervals()
+					.includeTopic(FIRST_TOPIC)
+					.doNotRetryOnDltFailure()
+					.dltHandlerMethod("myCustomDltProcessor", DLT_METHOD_NAME)
+					.create(template);
+		}
+
+		@Bean
+		public RetryTopicConfiguration secondRetryTopic(KafkaTemplate<String, String> template) {
+			return RetryTopicConfigurationBuilder
+					.newInstance()
+					.exponentialBackoff(500, 2, 10000)
+					.retryOn(Arrays.asList(IllegalStateException.class, IllegalAccessException.class))
+					.traversingCauses()
+					.includeTopic(SECOND_TOPIC)
+					.doNotRetryOnDltFailure()
+					.dltHandlerMethod("myCustomDltProcessor", DLT_METHOD_NAME)
+					.create(template);
+		}
+
+		@Bean
+		public FirstTopicListener firstTopicListener() {
+			return new FirstTopicListener();
+		}
+
+		@Bean
+		public KafkaListenerErrorHandler myCustomErrorHandler(CountDownLatchContainer container) {
+			return (message, exception) -> {
+				container.customErrorHandlerCountdownLatch.countDown();
+				throw exception;
+			};
+		}
+
+		@Bean
+		public SmartMessageConverter myCustomMessageConverter(CountDownLatchContainer container) {
+			return new CompositeMessageConverter(Collections.singletonList(new GenericMessageConverter())) {
+
+				@Override
+				public Object fromMessage(Message<?> message, Class<?> targetClass, Object conversionHint) {
+					container.customMessageConverterCountdownLatch.countDown();
+					return super.fromMessage(message, targetClass, conversionHint);
+				}
+			};
+		}
+
+		@Bean
+		public SecondTopicListener secondTopicListener() {
+			return new SecondTopicListener();
+		}
+
+		@Bean
+		public ThirdTopicListener thirdTopicListener() {
+			return new ThirdTopicListener();
+		}
+
+		@Bean
+		public FourthTopicListener fourthTopicListener() {
+			return new FourthTopicListener();
+		}
+
+		@Bean
+		public FifthTopicListener1 fifthTopicListener1() {
+			return new FifthTopicListener1();
+		}
+
+		@Bean
+		public FifthTopicListener2 fifthTopicListener2() {
+			return new FifthTopicListener2();
+		}
+
+		@Bean
+		SixthTopicDefaultDLTListener manualTopicListener() {
+			return new SixthTopicDefaultDLTListener();
+		}
+
+		@Bean
+		public NoRetryTopicListener noRetryTopicListener() {
+			return new NoRetryTopicListener();
+		}
+
+		@Bean
+		public FirstReuseRetryTopicListener firstReuseRetryTopicListener() {
+			return new FirstReuseRetryTopicListener();
+		}
+
+		@Bean
+		public SecondReuseRetryTopicListener secondReuseRetryTopicListener() {
+			return new SecondReuseRetryTopicListener();
+		}
+
+		@Bean
+		public ThirdReuseRetryTopicListener thirdReuseRetryTopicListener() {
+			return new ThirdReuseRetryTopicListener();
+		}
+
+		@Bean
+		CountDownLatchContainer latchContainer() {
+			return new CountDownLatchContainer();
+		}
+
+		@Bean
+		MyCustomDltProcessor myCustomDltProcessor() {
+			return new MyCustomDltProcessor();
+		}
+	}
+
+	@Configuration
+	public static class KafkaProducerConfig {
```

Comment:
public?

Reply:
Thanks for quick review. 
To many interesting unit test detail i need notice, thanks again!

Reply:
Thanks for your review, fix it.